### ISSUE TITLE
Update eslint-plugin-nx dependencies

### ISSUE THREAD

**Issue by t-ricci-delta** [nrwl/nx#8693]
- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md))

## Description

@nrwl/eslint-plugin-nx dependencies are stuck to version `~5.3.0` of packages `@typescript-eslint/parser` and `@typescript-eslint/experimental-utils`

Currently the latest NX workspace version (updated using `nx migrate latest`) will use Angular 13.1.1 and Typescript 4.5.4.

BUT `@typescript-eslint/parser` and `@typescript-eslint/experimental-utils` require Typescript < 4.5.0 and will output a warning when linting the project with `nx lint`

```Linting "project"...
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.5.0

YOUR TYPESCRIPT VERSION: 4.5.4

Please only submit bug reports when using the officially supported version.
```

Angular requires Typescript 4.5.x so it can't be downgraded and ESlint packages should be updated instead

## Motivation

Keep workspace libraries updated and eslint typecript parser and rules without warnings, that should work without issues

## Suggested Implementation
Update packages `@typescript-eslint/parser` and `@typescript-eslint/experimental-utils` to a version that supports typescript version >=4.5.x

**Comment by Phillip9587** [comment]
@JamesHenry Can you please take a look? It would be very nice to get rid of the warnings in our workspaces.

**Comment by meeroslav** [comment]
Hi @t-ricci-delta, thank you for reporting it.

This was already resolved last week by https://github.com/nrwl/nx/pull/8457 and should be available in 13.7.0+ versions.
